### DISCUSSION TITLE
IsaacEnv definition and TorchRL

### DISCUSSION THREAD

**Discussion by ADebor** [isaac-sim/IsaacLab#108] [Ideas]
Hi there, 

I was simply wondering if you had any updates on the potential migration of the base `IsaacEnv` as proposed [here](https://github.com/NVIDIA-Omniverse/Orbit/issues/59#issue-1665066762)? One of the ideas was to migrate the definition to the one used in [TorchRL](https://github.com/pytorch/rl) for [EnvBase](https://github.com/pytorch/rl/blob/main/torchrl/envs/common.py#L120). 

Being new to both Orbit and TorchRL and willing to use them together for my project, having similar interfaces on each side could simplify the integration (I guess). That being said, it's maybe not that complicated to interface these two libraries (I've tried a bit but not enough to call it complicated yet), and if someone has a script doing that already, I'd be more than happy to have a look at it. The TorchRL team just made an update providing IsaacGym integration, but as I got it, Orbit should be the preferred framework for future development involving robot learning and Isaac Sim.

**Comment by vmoens** [comment]
Happy to support this effort. We already have integrated Isaac gym envs (https://github.com/pytorch/rl/pull/1443) and we'd be thrilled to see more of these!
@ADebor would you like to give it a first shot? Should this be within torchrl or orbit?

**Reply by ADebor** [reply]
Hi @vmoens,

Glad to see you're interested in this. I've tried interfacing a bit, with varying degrees of success...

As for the location of the development, I'd rather see this in torchrl. In the Orbit doc, it says that "[...] to reduce complexity, we directly use the gym.Env over here and leave it up to library-defined wrappers to take care of wrapping this environment for their agents.". I suppose the proper way to do it would be to create an OrbitEnv class in torchrl (similar to IsaacGymEnv or IsaacGymWrapper, I'm not sure I understand the difference in scope between these two classes).  

If this sounds good, should I start an issue or a discussion in the torchrl repo to describe my use case and discuss what I've tried so far?

**Reply by vmoens** [reply]
Sorry for the very late reply
That sounds awesome to me!
